test/test_helper.rb

```diff
@@ -0,0 +1,114 @@
+$:.push(File.expand_path('../../plugins', __FILE__))
+$:.push(File.expand_path('../../lib', __FILE__))
+
+require 'bundler/setup'
+
+require 'vagrant-registration'
+require 'guests/redhat/cap/subscription_manager'
+require 'guests/redhat/cap/rhn_register'
+
+require 'minitest/autorun'
+require 'mocha/mini_test'
+
+require_relative 'support/fake_ui.rb'
+
+def fake_environment(options = { enabled: true })
+  { machine: fake_machine(options), ui: FakeUI, home_path: '/Users/hardy/work/developer-platform/vagrant/adb-vagrant-registration/build' }
+end
+
+class RecordingCommunicator
+  attr_reader :commands, :responses
+
+  def initialize
+    @commands = Hash.new([])
+    @responses = Hash.new('')
+  end
+
+  def stub_command(command, response)
+    responses[command] = response
+  end
+
+  def sudo(command)
+    #puts "SUDO: #{command}"
+    commands[:sudo] << command
+    responses[command]
+  end
+
+  def execute(command)
+    #puts "EXECUTE: #{command}"
+    commands[:execute] << command
+    responses[command].split("\n").each do |line|
+      yield(:stdout, "#{line}\n")
+    end
+  end
+
+  def test(command)
+    commands[:test] << command
+    true
+  end
+
+  def ready?
+    true
+  end
+end
+
+module Registration
+  class FakeProvider
+    def initialize(*args)
+    end
+
+    def _initialize(*args)
+    end
+
+    def ssh_info
+    end
+
+    def state
+      @state ||= Vagrant::MachineState.new('fake-state', 'fake-state', 'fake-state')
+    end
+  end
+end
+
+module Registration
+  class FakeConfig
+    def registration
+      @registration_config ||= VagrantPlugins::Registration::Config.new
+    end
+
+    def vm
+      VagrantPlugins::Kernel_V2::VMConfig.new
+    end
+  end
+end
+
+def fake_machine(options={})
+  env = options.fetch(:env, Vagrant::Environment.new({home_path: '/Users/hardy/work/developer-platform/vagrant/adb-vagrant-registration/build'}))
```

Comment:
I'll change that. It might even not be necessary anymore. Initially this was part of the attempts to determine why the tests would not properly work, but it turned out to be mainly a Rake issue.